Providers/Profile.swift

```diff
@@ -505,14 +510,58 @@ open class BrowserProfile: Profile {
     public func getCachedClients()-> Deferred<Maybe<[RemoteClient]>> {
         return self.remoteClientsAndTabs.getClients()
     }
+    
+    func getTabsWithNativeClients() -> Deferred<Maybe<[ClientAndTabs]>> {
+        // Because we are now using the application services tabs component with
+        // the iOS clients component (which has additional client data), we need
+        // to ensure that the clients we get from the tabs `getAll` call also
+        // exists in the clients BrowserDB table. This function will be obsolete
+        // once the sync manager component has been integrated into iOS and the
+        // iOS client synchronizer has been removed.
+        
+        return self.tabs.getAll().bind { tabsResult in
+            if let tabsError = tabsResult.failureValue {
+                return deferMaybe(tabsError)
+            }
+            
+            guard let clientRemoteTabs = tabsResult.successValue else {
+                return deferMaybe([])
+            }
+            
+            return self.remoteClientsAndTabs.getClients().bind { result in
+                if let error = result.failureValue {
+                    return deferMaybe(error)
+                }
+
+                guard let clients = result.successValue else {
+                    return deferMaybe([])
+                }
+
+                let clientAndTabs: [ClientAndTabs] = clientRemoteTabs.map { record in
+                    // We check if the application services clientId matches any client GUID.
+                    // If a client is found we return a record, otherwise we continue to the next
+                    // application services record.
+                    let localClient = clients.first(where: { $0.guid == record.clientId })
+
+                    if let client = localClient {
+                        return record.toClientAndTabs(client: client)
+                    }
+                    
```

Comment:
Do you think it's work logging that we failed to find the client for the tabs? Really not sure how useful logs are on iOS tbh, so possibly not!

Reply:
I'm going to add a debug logging for this case. I think sentry reporting _may_ be overkill at this point.